omnigibson/simulator.py

```diff
@@ -477,46 +478,50 @@ def import_object(self, obj, register=True):
             # Lastly, additionally add this object automatically to be initialized as soon as another simulator step occurs
             self.initialize_object_on_next_sim_step(obj=obj)
 
-        def remove_object(self, obj):
+        def remove_object(self, obj, has_registered=True):
             """
             Remove one or a list of non-robot object from the simulator.
 
             Args:
                 obj (BaseObject or Iterable[BaseObject]): one or a list of non-robot objects to remove
+                has_registered (bool): whether the object has been registered in the scene registry
             """
-            state = self.dump_state()
-
             objs = [obj] if isinstance(obj, BaseObject) else obj
 
-            # Omniverse has a strange bug where if GPU dynamics is on and the object to remove is in contact with
-            # with another object (in some specific configuration only, not always), the simulator crashes. Therefore,
-            # we first move the object to a safe location, then remove it.
-            pos = list(m.OBJECT_GRAVEYARD_POS)
-            for ob in objs:
-                ob.set_position_orientation(pos, [0, 0, 0, 1])
-                pos[0] += max(ob.aabb_extent)
+            if self.is_playing():
+                state = self.dump_state()
+
+                # Omniverse has a strange bug where if GPU dynamics is on and the object to remove is in contact with
+                # with another object (in some specific configuration only, not always), the simulator crashes. Therefore,
+                # we first move the object to a safe location, then remove it.
+                pos = list(m.OBJECT_GRAVEYARD_POS)
+                for ob in objs:
+                    ob.set_position_orientation(pos, [0, 0, 0, 1])
+                    pos[0] += max(ob.aabb_extent)
 
-            # One timestep will elapse
-            self.app.update()
+                # One timestep will elapse
+                self.app.update()
```

Comment:
Why this way instead of calling `self.step_physics` or if we dont want to clear APIs `self._physics_context._step(current_time=self.current_time)` like in `step_physics`? Just curious.

Reply:
It was `self.app.update()` before so I didn't think too much. But yeah, I think `self.step_physics()` makes more sense. Fixed.